keras/src/backend/tensorflow/trainer.py

```diff
@@ -621,6 +631,18 @@ def loss(self, y, y_pred, sample_weight=None):
             x=None, y=y, y_pred=y_pred, sample_weight=sample_weight
         )
 
+    def _symbolic_build(self, iterator=None, data_batch=None):
```

Comment:
I'd like to avoid any undesirable side effects as much as possible, and symbolic build may potentially have some. Please restrict this call to if the distribution strategy is set, so we don't change anything unless symbolic building is actually needed